src/clipboard_handlers/cell.ts

```diff
@@ -0,0 +1,289 @@
+import { CommandResult } from "..";
+import { formatValue } from "../helpers";
+import { pastedZones } from "../helpers/clipboard/clipboard_helpers";
+import {
+  CellPosition,
+  ClipboardCell,
+  ClipboardData,
+  ClipboardOptions,
+  ClipboardPasteTarget,
+  HeaderIndex,
+  UID,
+  Zone,
+} from "../types";
+import { ClipboardHandler } from "./abstract_clipboard_handler";
+
+interface ClipboardContent {
+  cells: ClipboardCell[][];
+  zones: Zone[];
+  sheetId: UID;
+  isCutOperation: boolean;
+}
+
+export class CellClipboardHandler extends ClipboardHandler<ClipboardContent> {
+  isCutAllowed(data: ClipboardData) {
+    if ("zones" in data && data.zones.length !== 1) {
+      return CommandResult.WrongCutSelection;
+    }
+    return CommandResult.Success;
+  }
+
+  copy(isCutOperation: boolean, data: ClipboardData): ClipboardContent | undefined {
+    if (!("zones" in data) || !data.zones.length) {
+      return;
+    }
+    const sheetId = this.getters.getActiveSheetId();
+    const zones = data.zones;
+    if (!zones.length) {
+      return {
+        cells: [[]],
+        zones: [],
+        isCutOperation,
+        sheetId,
+      };
+    }
+
+    const { clippedZones, rowsIndex, columnsIndex } = data;
+    const clippedCells: ClipboardCell[][] = [];
+
+    for (let row of rowsIndex) {
+      let cellsInRow: ClipboardCell[] = [];
+      for (let col of columnsIndex) {
+        const position = { col, row, sheetId };
+        const spreader = this.getters.getArrayFormulaSpreadingOn(position);
+        let cell = this.getters.getCell(position);
+        const evaluatedCell = this.getters.getEvaluatedCell(position);
+        if (spreader) {
+          const isSpreaderCopied =
+            rowsIndex.includes(spreader.row) && columnsIndex.includes(spreader.col);
+          const content = isSpreaderCopied
+            ? ""
+            : formatValue(evaluatedCell.value, { locale: this.getters.getLocale() });
+          cell = {
+            id: cell?.id || "",
+            style: cell?.style,
+            format: evaluatedCell.format,
+            content,
+            isFormula: false,
+          };
+        }
+        cellsInRow.push({
+          cell,
+          border: this.getters.getCellBorder(position) || undefined,
+          evaluatedCell,
+          position,
+        });
+      }
+      clippedCells.push(cellsInRow);
+    }
+
+    return {
+      cells: clippedCells,
+      zones: clippedZones,
+      sheetId: this.getters.getActiveSheetId(),
+      isCutOperation,
+    };
+  }
+
+  isPasteAllowed(
+    sheetId: UID,
+    target: Zone[],
+    content: ClipboardContent,
+    clipboardOption?: ClipboardOptions
+  ): CommandResult {
+    if (!("cells" in content)) {
+      return CommandResult.Success;
+    }
+    if (content.isCutOperation && clipboardOption?.pasteOption !== undefined) {
+      // cannot paste only format or only value if the previous operation is a CUT
+      return CommandResult.WrongPasteOption;
+    }
```

Comment:
So we can paste format only from a cut operation if there is no cells in the copy area ? Something seems strange here ...

Reply:
I'm not sure about what you mean. The "if !cells..." is there to check if we have copied cells or figure. Where there is no cell in the content, we won't call the paste of cell.ts, so we won't copy any format.